### PR TITLE
feat: create new dropdown badge

### LINKED ISSUE #2830: Renaming C7/C8 – Create new file dropdown
__What should we do?__

We should implement the new file dropdown as per https://github.com/camunda/camunda-modeler-design/issues/72

This includes:

- [x] Changed order between C7 and C8 sections
- [x] Section rename
- ~~New DMN button~~ This requires an entire implementation in the tabs provider and does not make sense under my scope, so @barmac will add that as part of his implementation of cloud DMN 
- [x] "New" badge on C8

__Why should we do it?__

To improve visibility on C8 and hopefully drive adoption.

Epic: https://github.com/bpmn-io/internal-docs/issues/490

### PR DESCRIPTION
**Pull request by Skaiir**
Closes #2830 

Screenshot:
image

### CODE REVIEW

**Review by andreasgeier — APPROVED** [review]
✅  Looks as expected!

File: client/src/app/App.js [hunk 1]
```diff
@@ -2007,11 +2009,18 @@ export class App extends PureComponent {
...
+      var item = {
         items: group,
         key,
-        label: key
+        label: key,
       };
+
+      if (key === 'Camunda Platform 8') {
+        item.labelSuffix = <NewBadge inline style={ { marginLeft: '20px', transform: 'translate(0px, -1px)' } } />;
+      }
```
Comment (pinussilvestrus) [inline client/src/app/App.js:2021]: What about having a simple test case for `_getNewFileItems` whether it returns a label suffix for C8? Or is it so temporarily that we don't need it?

**Review by pinussilvestrus — CHANGES_REQUESTED** [review]
I like 👍  Let's consider some test coverage (or make an active decision that we don't need some)

File: client/src/app/App.js (hunk 1 above)
Comment (Skaiir) [reply client/src/app/App.js:2021]: No I agree with you I'll do that.

File: client/src/app/App.js [hunk 2]
```diff
@@ -2007,11 +2009,18 @@ export class App extends PureComponent {
     });
 
     const groupedItems = map(groupBy(items, 'group'), (group, key) => {
-      return {
+
+      var item = {
```
Comment (pinussilvestrus) [inline client/src/app/App.js:2013]:
```suggestion
      let item = {
```

File: client/src/app/__tests__/AppSpec.js [hunk 3]
```diff
@@ -308,6 +308,24 @@ describe('<App>', function() {
       expect(activeTab).to.eql(tabs[ tabs.length - 1 ]);
     });
 
+    it('should include labelSuffix for Camunda Platform 8', function() {
```
Comment (pinussilvestrus) [inline client/src/app/__tests__/AppSpec.js:311]: Is `diagramCreation` really the right place? Why not just a new `describe('#_getNewFileitems')`?

File: client/src/app/__tests__/AppSpec.js [hunk 4]
```diff
@@ -308,6 +308,24 @@ describe('<App>', function() {
...
+
+      // when
+      const fileItems = app._getNewFileItems();
+
+      // then
+      const platformGroup = fileItems[0];
+      const cloudGroup = fileItems[1];
+
+      expect(platformGroup.labelSuffix).is.undefined;
+      expect(cloudGroup.labelSuffix).is.not.undefined;
```
Comment (pinussilvestrus) [inline client/src/app/__tests__/AppSpec.js:327]: Would it also accept if it is `null`? Therefore I'd just do

```js
expect(platformGroup.labelSuffix).to.not.exist;
expect(cloudGroup.labelSuffix).to.exist;
```

**Review by pinussilvestrus — APPROVED** [review]
🍯

**Pull request merged**